ppr-ui/src/components/mhrTransfers/LocationChange.vue

```diff
@@ -103,48 +136,126 @@
-import { toRefs, reactive, computed, watch } from "vue"
+import { reactive, computed, watch, ref, onMounted, nextTick } from "vue"
 import { FormCard } from "../common"
 import { HomeCivicAddress, HomeLandOwnership, HomeLocationType } from "../mhrRegistration"
 import { CivicAddressSchema } from '@/schemas/civic-address'
 import { TaxCertificate } from "."
+import { useInputRules } from "@/composables/useInputRules"
+import { useMhrInfoValidation, useTransportPermits } from "@/composables"
+import { storeToRefs } from "pinia"
+import { changeTransportPermitLocationTypeDialog } from '@/resources/dialogOptions'
+import { BaseDialog } from '@/components/dialogs'
+import { cloneDeep } from 'lodash'
 
-defineProps<{
+const props = defineProps<{
+  validate: boolean,
   content?: ContentIF
 }>()
 
-const emit = defineEmits(['updateLocationType'])
+const { isRoleQualifiedSupplier, getMhrRegistrationLocation,
+  setMhrTransportPermit, setMhrTransportPermitNewLocation } = useStore()
+
+const { hasUnsavedChanges, getMhrTransportPermit, getMhrInfoValidation } = storeToRefs(useStore())
+
+const { setLocationChangeType, initTransportPermit, resetTransportPermit } = useTransportPermits()
+
+const {
+  setValidation,
+  getInfoValidation,
+  resetValidationState
+} = useMhrInfoValidation(getMhrInfoValidation.value)
+
+const { required } = useInputRules()
 
-const {  } = toRefs(useStore())
+const locationChangeForm = ref(null) as FormIF
+const locationChangeSelectRef = ref(null) as FormIF
 
-const { isRoleQualifiedSupplier, setMhrLocation, getMhrRegistrationLocation } = useStore()
+const homeCivicAddressRef = ref(null)
+const taxCertificateRef = ref(null)
 
 const state = reactive({
-  locationChangeType: null,
+  locationChangeType: getMhrTransportPermit.value.locationChangeType,
+  prevLocationChangeType: null,
+  locationChangeFromValid: false,
   roleBasedLocationChangeTypes: computed(() =>
     isRoleQualifiedSupplier
       ? locationChangeTypes.slice(0, -1) // qualified supplier does not have the third option in menu
       : locationChangeTypes),
   isTransportPermitType: computed(() => state.locationChangeType === LocationChangeTypes.TRANSPORT_PERMIT),
-  isNotManufacturersLot: computed(() => getMhrRegistrationLocation.locationType !== HomeLocationTypes.LOT)
+  isNotManufacturersLot: computed(() => getMhrRegistrationLocation.locationType !== HomeLocationTypes.LOT),
+  showChangeTransportPermitLocationTypeDialog: false
+})
+
+onMounted(async (): Promise<void> => {
+  initTransportPermit()
 })
 
 const handleTaxCertificateUpdate = (date: string) => {
-  setMhrLocation({ key: 'taxExpiryDate', value: date })
-  setMhrLocation({ key: 'taxCertificate', value: !!date })
+  setMhrTransportPermitNewLocation({ key: 'taxExpiryDate', value: date })
+  setMhrTransportPermitNewLocation({ key: 'taxCertificate', value: !!date })
 }
 
 const handleLocationTypeUpdate = (newLocation: { key, value }) => {
-  console.log(newLocation);
```

Comment:
Thank you 👍

Reply:
Yea, this was a placeholder until getters/setters were ready 👌🏻